R/FilteredData.R

```diff
@@ -904,6 +972,12 @@ FilteredData <- R6::R6Class( # nolint
     # private attributes ----
     filtered_datasets = list(),
 
+    # turn on / off filter panel
+    filter_turn = TRUE,
```

Comment:
(copied from closed PR as I still think it's relevant)

1. I dislike the "turn" term here. You can _turn_ left/right or on/off. It's a verb whereas you are looking for a noun. I propose _state_ / _status_ / _active_ instead. Please change other places as well including method docs.

2. What's the point of this field? I ask from the class simplicity point of view.
I can see that it's used in the api class - there is a check whether it's active and if yes then you can do set/get/etc. Is that needed? The functionality would be anyway disabled from the GUI perspective (vide: disabling inputs). What I am suggesting here is that the big disable/enable button would _just_ propagate this action to all relevant inputs + save/restore action and that's it. An object would not store its state as this opens a door for unnecessary complexity I believe. I am happy to discuss it further if needed.

_Originally posted by @pawelru in https://github.com/insightsengineering/teal.slice/pull/91#discussion_r961522114_

----

By default `shinyWidgets::switchInput` use the logical(1) for value so this should not be surprising I stay with this type.

`filter_turn` variable is needed only for filter_panel_api, so we could provide a proper warning if sb want to update the state when filter panel is turn off.

I like the Turn Off/On naming, although I see you agree with Nik in this area so please update to the one you are preferring.

_Originally posted by @Polkas in https://github.com/insightsengineering/teal.slice/pull/91#discussion_r961543287_

Reply:
1. I am ok with the active naming convention.
2. I feel you need a little bit of background and support. I am happy to help you here.

Reply:
@pawelru you not answer to more important question which is if we want to support filter panel api at all or in what scope.